bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/discovery/NetatmoDiscoveryService.java

```diff
@@ -0,0 +1,189 @@
+import static org.openhab.binding.netatmo.internal.NetatmoBindingConstants.EQUIPMENT_ID;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.netatmo.internal.api.AircareApi;
+import org.openhab.binding.netatmo.internal.api.ApiBridge;
+import org.openhab.binding.netatmo.internal.api.ConnectionListener;
+import org.openhab.binding.netatmo.internal.api.ConnectionStatus;
+import org.openhab.binding.netatmo.internal.api.ModuleType;
+import org.openhab.binding.netatmo.internal.api.NetatmoConstants.FeatureArea;
+import org.openhab.binding.netatmo.internal.api.NetatmoException;
+import org.openhab.binding.netatmo.internal.api.WeatherApi;
+import org.openhab.binding.netatmo.internal.api.dto.NAHome;
+import org.openhab.binding.netatmo.internal.api.dto.NAMain;
+import org.openhab.binding.netatmo.internal.api.dto.NAMain.NAStationDataResponse;
+import org.openhab.binding.netatmo.internal.api.dto.NAThing;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.i18n.LocaleProvider;
+import org.openhab.core.i18n.TranslationProvider;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NetatmoDiscoveryService} searches for available Netatmo
+ * devices and modules connected to the API console
+ *
+ * @author Gaël L'hopital - Initial contribution
+ *
+ */
+@Component(service = DiscoveryService.class, configurationPid = "binding.netatmo")
+@NonNullByDefault
+public class NetatmoDiscoveryService extends AbstractDiscoveryService implements ConnectionListener {
+    private static final int DISCOVER_TIMEOUT_SECONDS = 5;
+    private final Logger logger = LoggerFactory.getLogger(NetatmoDiscoveryService.class);
+    private final ApiBridge apiBridge;
+
+    @Activate
+    public NetatmoDiscoveryService(@Reference ApiBridge apiBridge, @Reference LocaleProvider localeProvider,
+            @Reference TranslationProvider translationProvider) {
+
+        super(ModuleType.asSet.stream().filter(mt -> mt != ModuleType.UNKNOWN).map(mt -> mt.thingTypeUID)
+                .collect(Collectors.toSet()), DISCOVER_TIMEOUT_SECONDS);
+        this.apiBridge = apiBridge;
+        this.localeProvider = localeProvider;
+        this.i18nProvider = translationProvider;
+        apiBridge.addConnectionListener(this);
+    }
+
+    @Override
+    public void apiConnectionStatusChanged(ConnectionStatus connectionStatus) {
+        if (connectionStatus.equals(ConnectionStatus.SUCCESS)) {
```

Comment:
To eliminate risk of NullPointerException:
`if (ConnectionStatus.SUCCESS.equals(connectionStatus)) {`

Reply:
Corrected.